### DISCUSSION TITLE
Feature Request: Separate S3 API multipart upload finalising with COPY parameter

### DISCUSSION THREAD

**Discussion by dforsber** [duckdb/duckdb#11643] [Ideas]
Is there a way to separate S3 multipart upload and its completion with DuckDB? Like having COPY output parameter stating whether to complete the upload or not? This would be killer feature for data pipelines producing smaller chunks of data per iteration while reducing the need for post compaction processing.

For example, a new boolean parameter `S3_MULTIPART_FINALISE`  for COPY to S3 targets.

```sql
COPY ( FROM read_csv_auto('chunk1') ) 
TO 's3://bucket/all_chunks.parquet' 
WITH ( FORMAT 'Parquet', COMPRESSION 'zstd', S3_MULTIPART_FINALISE=false)

COPY ( FROM read_csv_auto('chunk2') ) 
TO 's3://bucket/all_chunks.parquet' 
WITH ( FORMAT 'Parquet', COMPRESSION 'zstd', S3_MULTIPART_FINALISE=false)

COPY ( FROM read_csv_auto('chunk3') ) 
TO 's3://bucket/all_chunks.parquet' 
WITH ( FORMAT 'Parquet', COMPRESSION 'zstd', S3_MULTIPART_FINALISE=true)
```

**Comment by rustyconover** [comment]
You can't really append Parquet files since there is important information stored in the footer of the Parquet file that points to the row groups, but I could see this being possible with CSV/JSON.  But you also run into problems with where do you track all of the already created upload parts and know that you're adding parts in a proper order.

**Comment by dforsber** [comment]
It's not appending, it's in-progress, and yes, you need to keep state, both S3 ETags and ids, but also the Parquet metadata. DuckDB already streams to S3 with multi-part upload... just extend it a bit ;).
